### PR TITLE
remove obsolete console stuff, new logging project

### PR DESCRIPTION
**Pull request by amazingalek**
* removed obsolete console stuff
* migrated logging stuff to new project
* static console for OWML

Tested with all mods, except NomaiVR.

### CODE REVIEW

File: OWML.ModHelper/Logging/ModConsole.cs [hunk 1]
```diff
@@ -26,15 +23,10 @@ protected ModConsole(IOwmlConfig config, IModLogger logger, IModManifest manifes
...
+            if (Manifest.Name == Constants.OwmlTitle)
             {
-                Instance = this;
+                OwmlConsole = this;
             }
         }
-
-        internal static void CallWriteCallback(IModManifest manifest, string text)
-        {
-            OnConsole?.Invoke(manifest, text);
```
Comment (misternebula) [inline OWML.ModHelper/Logging/ModConsole.cs:37]: uh, where has this gone...?

Comment (misternebula) [reply OWML.ModHelper/Logging/ModConsole.cs:37]: would break backwards compatibility

Comment (Raicuparta) [reply OWML.ModHelper/Logging/ModConsole.cs:37]: Pretty sure that's intended? 😄

File: OWML.Launcher/Program.cs [hunk 2]
```diff
@@ -37,6 +44,7 @@ private static void SaveConsolePort(IOwmlConfig owmlConfig)
                     return;
                 }
                 owmlConfig.SocketPort = port;
+                // todo shouldn't we actually **save** too?
```
Comment (misternebula) [inline OWML.Launcher/Program.cs:47]: ah yeah, forgot about that 🥴

Comment (misternebula) [reply OWML.Launcher/Program.cs:47]: saving is done in socketlistener, so just copy the json line to here

File: OWML.ModHelper/Logging/ModConsole.cs (hunk 1 above)
Comment (amazingalek) [reply OWML.ModHelper/Logging/ModConsole.cs:37]: Yup, the static event is what stopped us from migrating logging/console stuff to new project. It's intended :)

Comment (TAImatem) [reply OWML.ModHelper/Logging/ModConsole.cs:37]: it's `internal` anyway, therefore should not break compatibility

**Comment by amazingalek** [comment]
I'll merge #244 (after it's approved) into this before merging to dev.

File: OWML.Launcher/Program.cs [hunk 3]
```diff
@@ -37,6 +45,7 @@ private static void SaveConsolePort(IOwmlConfig owmlConfig)
                     return;
                 }
                 owmlConfig.SocketPort = port;
+                JsonHelper.SaveJsonObject(Constants.OwmlConfigFileName, owmlConfig);
```
Comment (amazingalek) [inline OWML.Launcher/Program.cs:48]: fixed an unrelated oversight

File: OWML.Launcher/Program.cs [hunk 4]
```diff
@@ -26,9 +27,16 @@ static void Main(string[] args)
             app.Run(args);
         }
 
-        private static void SaveConsolePort(IOwmlConfig owmlConfig)
+        private static IModConsole CreateWriter(IOwmlConfig owmlConfig, IModManifest owmlManifest, bool hasConsolePort)
```
Comment (amazingalek) [inline OWML.Launcher/Program.cs:30]: This method replaces OutputFactory

File: OWML.Launcher/Program.cs [hunk 5]
```diff
@@ -67,5 +69,12 @@ private static IModManifest GetOwmlManifest()
             return JsonHelper.LoadJsonObject<ModManifest>(Constants.OwmlManifestFileName);
         }
 
+        private static IModConsole CreateWriter(IOwmlConfig owmlConfig, IModManifest owmlManifest, bool hasConsolePort)
```
Comment (amazingalek) [inline OWML.Launcher/Program.cs:72]: This method replaces OutputFactory.

**Pull request merged**